vnext/Shared/HermesSamplingProfiler.cpp

```diff
@@ -4,25 +4,42 @@
 #include "pch.h"
 
 #include <chrono>
+#include <future>
 
 #include <hermes/hermes.h>
-#include <winrt/Windows.Storage.h>
 #include "Unicode.h"
 
+#ifdef WINRT
+#include <winrt/Windows.Storage.h>
+#else
+#include <ShlObj.h>
+#include <Shlwapi.h>
+#endif
+
 #include "HermesSamplingProfiler.h"
 
 namespace Microsoft::ReactNative {
 
 namespace {
 std::future<std::string> getTraceFilePath() noexcept {
+  std::ostringstream os;
+#ifdef WINRT
   auto hermesFolder = (co_await winrt::Windows::Storage::ApplicationData::Current().LocalFolder().CreateFolderAsync(
                            L"Hermes", winrt::Windows::Storage::CreationCollisionOption::OpenIfExists))
                           .Path();
+  os << Microsoft::Common::Unicode::Utf16ToUtf8(hermesFolder.c_str(), hermesFolder.size());
+#else
+  WCHAR wzAppData[MAX_PATH];
+  if (SHGetFolderPathW(nullptr, CSIDL_LOCAL_APPDATA, nullptr, SHGFP_TYPE_CURRENT, wzAppData) != S_OK)
+    std::abort(); // We don't expect this to happen ever ..
+
+  std::string appData = Microsoft::Common::Unicode::Utf16ToUtf8(wzAppData, wcsnlen_s(wzAppData, MAX_PATH));
```

Comment:
nit, you can simplify a bit (and use standard type):
```suggestion
  wchar_t wzAppData[MAX_PATH]{};
  if (SHGetFolderPathW(nullptr, CSIDL_LOCAL_APPDATA, nullptr, SHGFP_TYPE_CURRENT, wzAppData) != S_OK)
    std::abort(); // We don't expect this to happen ever ..

  std::string appData = winrt::to_string(wzAppData);
```

Reply:
Removing the conditional compilation results in WAC failure ! hence i reverted to conditional compilation .. Relying on the same WINRT macro for now as it's being used at other places too ..

Reply:
WINRT will still be defined though so I'm not sure that is the right fix. If WACK is failing because of the SHGetFolderPathW call, you can work around it by using `CallIndirect<SHGetFolderPathW>` (see https://github.com/microsoft/react-native-windows/blob/003602b90aeda1b2a43702c6a9307c0f01010e59/vnext/Microsoft.ReactNative.Cxx/DesktopWindowBridge.h)